05.wc/wcc.rb

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+WIDTH_SUM_VALUE = 8
+
+def main(args)
+  options = {}
+  options[:filepaths] = args[:filepaths] || args[:filepaths] = false
+  options[:l]         = args[:l] || args[:l] = false
+  options[:w]         = args[:w] || args[:w] = false
+  options[:c]         = args[:c] || args[:c] = false
+
+  if options[:filepaths].empty?
+    show_totals_pipe_input(options)
+  else
+    show_totals_file_input(options)
+  end
+end
+
+def show_totals_pipe_input(options)
+  input_totals = count_input(readlines.join)
+  show_result(input_totals, options)
+end
+
+def show_totals_file_input(options)
+  input_totals_list = []
+  options[:filepaths].each do |fp|
+    input_totals = count_input(File.open(fp).readlines.join)
+    show_result(input_totals, options, fp)
+    input_totals_list << input_totals
+  end
+  show_total(input_totals_list, options) if input_totals_list.size > 1
+end
+
+def count_input(input_readline)
+  {
+    lines_count: input_readline.split("\n").size,
+    words_count: input_readline.split("\s").size,
+    bytes_count: input_readline.bytesize
+  }
+end
+
+def show_result(count_results, options, counted_target = nil)
+  showed_line = ''
+  showed_line += adjust_show_width(count_results[:lines_count]) if show_lines?(options)
+  showed_line += adjust_show_width(count_results[:words_count]) if show_words?(options)
+  showed_line += adjust_show_width(count_results[:bytes_count]) if show_bytes?(options)
+  showed_line += " #{counted_target}" unless counted_target.nil?
+  puts showed_line
+end
+
+def show_total(input_totals_list, options)
+  count_results_sum = {
```

Comment:
totalの値を集めた配列、という意味で、

```suggestion
  totals = {
```

にするとか。（`count_results_sum`＝カウントした結果の合計？だとちょっと冗長な印象）

Reply:
承知しました。ご指摘のとおり修正しました